File: SchoolWebApp-API/Project.Core/Entities/Class/ClassLeadershipRole.cs
using SchoolWebApp.Core.Entities.Enums;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.Class
{
    public class ClassLeadershipRole: SettingsBase
    {
        public PersonType PersonType { get; set; }
        public List<SchoolClassLeaders> SchoolClassLeaders { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/class/models/class-leadership-role.ts
import {PersonType} from '@/core/enums/personTypes';
import {ResourceModel} from '@/core/models/ResourceModel';

export class ClassLeadershipRole extends ResourceModel<ClassLeadershipRole> {
    public name?: string;
    public description?: string;
    public personType?: PersonType;
    public rank?: number;

    constructor(model?: Partial<ClassLeadershipRole>) {
        super(model);
    }
}
